File: src/main/java/ti/proyectoinia/dtos/TetrazolioDto.java
package ti.proyectoinia.dtos;

import lombok.Data;
import lombok.Generated;
import ti.proyectoinia.business.entities.PreTratamiento;
import ti.proyectoinia.business.entities.ViabilidadPorTz;
import ti.proyectoinia.business.entities.ViabilidadVigorTZ;

import java.util.Date;

@Data
public class TetrazolioDto {
    
    private Long id;
    
    private Integer repeticion;
    
    private Integer nroSemillasPorRepeticion;
    
    private PreTratamiento pretratamiento;
    
    private String concentracion;
    
    private String tincionHoras;
    
    private String tincionGrados;
    
    private Date fecha;
    
    // Segundo conjunto de datos (para tabla de detalles)
    private Integer nroSemillasPorRepeticion2;
    
    private PreTratamiento pretratamiento2;
    
    private String concentracion2;
    
    private String tincionHoras2;
    
    private String tincionGrados2;
    
    private Date fecha2;
    
    private String viables;
    
    private String noViables;
    
    private String duras;
    
    private String total;
    
    private String promedio;
    
    private Integer porcentaje;
    
    private ViabilidadPorTz viabilidadPorTetrazolio;
    
    private Integer nroSemillas;
    
    private Integer daniosNroSemillas;
    
    private Integer daniosMecanicos;
    
    private Integer danioAmbiente;
    
    private Integer daniosChinches;
    
    private Integer daniosFracturas;
    
    private Integer daniosOtros;
    
    private Integer daniosDuras;
    
    private ViabilidadVigorTZ viabilidadVigorTz;
    
    private Integer porcentajeFinal;
    
    private Integer daniosPorPorcentajes;

    private boolean activo;

    private boolean estandar;

    private boolean repetido;

    private Date fechaCreacion;

    private Date fechaRepeticion;

    private Long reciboId;

    private ReporteTetrazolioDto reporte;

    @Generated
    public boolean equals(final Object o) {
        if (o == this) return true;
        if (!(o instanceof TetrazolioDto)) return false;
        TetrazolioDto other = (TetrazolioDto) o;
        return id != null && id.equals(other.getId());
    }

    @Generated
    public int hashCode() {
        return id != null ? id.hashCode() : 0;
    }

    @Generated
    public String toString() {
        return "TetrazolioDto(tetrazolioId=" + id + ", repeticion=" + repeticion + ", porcentajeFinal=" + porcentajeFinal + ")";
    }
}

File: frontend/src/models/Tetrazolio.dto.ts
export type ViabilidadPorTz = 'ALTA' | 'MEDIA' | 'BAJA';
export type ViabilidadVigorTZ = 'VIGOR_ALTO' | 'VIGOR_MEDIO' | 'VIGOR_BAJO' | 'LIMITE_CRITICO';

export interface PreTratamiento {
  id: number;
  nombre: string;
}

export interface ReporteFilaDanios {
  mecanicos: number | null;
  ambiente: number | null;
  chinches: number | null;
  fracturas: number | null;
  otros: number | null;
  duras: number | null;
}

export interface ReporteFila {
  porcentaje: number | null;
  danios: ReporteFilaDanios;
}

export interface ReporteTetrazolio {
  vigorAlto: ReporteFila;
  vigorMedio: ReporteFila;
  vigorBajo: ReporteFila;
  limiteCritico: ReporteFila;
  noViables: ReporteFila;
  viabilidad: ReporteFila;
  vigorAcumulado: ReporteFila;
}

export interface TetrazolioDto {
  id: number | null;
  repeticion: number | null;
  nroSemillasPorRepeticion: number | null;
  pretratamiento: PreTratamiento | null;
  concentracion: string | null;
  tincionHoras: string | null;
  tincionGrados: string | null;
  fecha: string | null;
  viables: string | null;
  noViables: string | null;
  duras: string | null;
  total: string | null;
  promedio: string | null;
  porcentaje: number | null;
  viabilidadPorTetrazolio: ViabilidadPorTz | null;
  nroSemillas: number | null;
  daniosNroSemillas: number | null;
  daniosMecanicos: number | null;
  danioAmbiente: number | null;
  daniosChinches: number | null;
  daniosFracturas: number | null;
  daniosOtros: number | null;
  daniosDuras: number | null;
  viabilidadVigorTz: ViabilidadVigorTZ | null;
  porcentajeFinal: number | null;
  daniosPorPorcentajes: number | null;
  activo: boolean;
  estandar: boolean;
  repetido: boolean;
  fechaCreacion: string | null;
  fechaRepeticion: string | null;
  reciboId: number | null;
  reporte: ReporteTetrazolio | null;
}
